### PR TITLE
Added auto_last_hop parameter for bigip_virtual_server module

### LINKED ISSUE #2104: bigip_virtual_server - auto last hop
### Is your feature request related to a problem? Please describe.
Using 'bigip_virtual_server' module is not possible change 'auto-lasthop' setting. There is no parameter for this.

'auto-lasthop' feature returns traffic based on sender MAC address (L2 address of the L3 address/gateway that sent packet), not based on routing table (next hop based on L3 address). In some cases is this behavior wrong - e.g. in my case with Cisco ACI. 

Cisco ACI, a.k.a. software-defined data center, manipulates mac addresses for internal communication between End Point Groups. If there is L4-L7 device (in this case 'load-balancer'), 'auto-lasthop' feature returns traffic to wrong mac address - sender's MAC (incoming traffic) and receiver's MAC address (outgoing traffic) is different even though, that L3 address is the same.

Documentation for the auto-lasthop setting: https://support.f5.com/csp/article/K13876

#### Workaround-1:
- configure virtual server using ansible
- then change manualy auto-lasthop using GUI/TMOS configuration

#### Workaround-2:
- first: change auto-lasthop globaly to disabled (using GUI/TMOS)
- then configure virtual server using ansible (no auto-lasthop change for virtual server is needed, becasue default setting is - use global setting)

^^ both workarounds needs some 'manual' change on the BIGIP side.

### Describe the solution you'd like
Add please parameter to change auto-lasthop for bigip_virtual_server module.

### PR DESCRIPTION
**Pull request by urohit011**
Fixes #2104 

Tests:

(venv-3.9) ➜  repro$ ansible-playbook -i inventory/hosts ../f5-ansible/test/integration/bigip_virtual_server.yaml -t "issue-02104" --step

PLAY [Metadata of bigip_virtual_server] ******************************************************************************************************************************************

PLAY [Test the bigip_virtual_server module - Environment] ************************************************************************************************************************
Perform task: TASK: Gathering Facts (N)o/(y)es/(c)ontinue: y

Perform task: TASK: Gathering Facts (N)o/(y)es/(c)ontinue: ***********************************************************************************************************************

TASK [Gathering Facts] ***********************************************************************************************************************************************************
ok: [bigip1]
Perform task: TASK: bigip_virtual_server : Issue 02104 - Add virtual server (N)o/(y)es/(c)ontinue: y

Perform task: TASK: bigip_virtual_server : Issue 02104 - Add virtual server (N)o/(y)es/(c)ontinue: *******************************************************************************

TASK [bigip_virtual_server : Issue 02104 - Add virtual server] *******************************************************************************************************************
changed: [bigip1]
Perform task: TASK: bigip_virtual_server : Issue 02104 - Change auto last hop to enabled (N)o/(y)es/(c)ontinue: y

Perform task: TASK: bigip_virtual_server : Issue 02104 - Change auto last hop to enabled (N)o/(y)es/(c)ontinue: ******************************************************************

TASK [bigip_virtual_server : Issue 02104 - Change auto last hop to enabled] ******************************************************************************************************
changed: [bigip1]
Perform task: TASK: bigip_virtual_server : Issue 02104 - Assert Change auto last hop to enabled (N)o/(y)es/(c)ontinue: y

Perform task: TASK: bigip_virtual_server : Issue 02104 - Assert Change auto last hop to enabled (N)o/(y)es/(c)ontinue: ***********************************************************

TASK [bigip_virtual_server : Issue 02104 - Assert Change auto last hop to enabled] ***********************************************************************************************
ok: [bigip1] => {
    "changed": false,
    "msg": "All assertions passed"
}
Perform task: TASK: bigip_virtual_server : Issue 02104 - Change auto last hop to enabled - Idempotent Check (N)o/(y)es/(c)ontinue: y

Perform task: TASK: bigip_virtual_server : Issue 02104 - Change auto last hop to enabled - Idempotent Check (N)o/(y)es/(c)ontinue: ***********************************************

TASK [bigip_virtual_server : Issue 02104 - Change auto last hop to enabled - Idempotent Check] ***********************************************************************************
ok: [bigip1]
Perform task: TASK: bigip_virtual_server : Issue 02104 - Assert Change auto last hop to enabled - Idempotent Check (N)o/(y)es/(c)ontinue: y

Perform task: TASK: bigip_virtual_server : Issue 02104 - Assert Change auto last hop to enabled - Idempotent Check (N)o/(y)es/(c)ontinue: ****************************************

TASK [bigip_virtual_server : Issue 02104 - Assert Change auto last hop to enabled - Idempotent Check] ****************************************************************************
ok: [bigip1] => {
    "changed": false,
    "msg": "All assertions passed"
}
Perform task: TASK: bigip_virtual_server : Issue 02104 - Change auto last hop to disabled (N)o/(y)es/(c)ontinue: y

Perform task: TASK: bigip_virtual_server : Issue 02104 - Change auto last hop to disabled (N)o/(y)es/(c)ontinue: *****************************************************************

TASK [bigip_virtual_server : Issue 02104 - Change auto last hop to disabled] *****************************************************************************************************
changed: [bigip1]
Perform task: TASK: bigip_virtual_server : Issue 02104 - Assert Change auto last hop to disabled (N)o/(y)es/(c)ontinue: y

Perform task: TASK: bigip_virtual_server : Issue 02104 - Assert Change auto last hop to disabled (N)o/(y)es/(c)ontinue: **********************************************************

TASK [bigip_virtual_server : Issue 02104 - Assert Change auto last hop to disabled] **********************************************************************************************
ok: [bigip1] => {
    "changed": false,
    "msg": "All assertions passed"
}
Perform task: TASK: bigip_virtual_server : Issue 02104 - Change auto last hop to disabled - Idempotent Check (N)o/(y)es/(c)ontinue: y

Perform task: TASK: bigip_virtual_server : Issue 02104 - Change auto last hop to disabled - Idempotent Check (N)o/(y)es/(c)ontinue: **********************************************

TASK [bigip_virtual_server : Issue 02104 - Change auto last hop to disabled - Idempotent Check] **********************************************************************************
ok: [bigip1]
Perform task: TASK: bigip_virtual_server : Issue 02104 - Assert Change auto last hop to disabled - Idempotent Check (N)o/(y)es/(c)ontinue: y

Perform task: TASK: bigip_virtual_server : Issue 02104 - Assert Change auto last hop to disabled - Idempotent Check (N)o/(y)es/(c)ontinue: ***************************************

TASK [bigip_virtual_server : Issue 02104 - Assert Change auto last hop to disabled - Idempotent Check] ***************************************************************************
ok: [bigip1] => {
    "changed": false,
    "msg": "All assertions passed"
}
Perform task: TASK: bigip_virtual_server : Issue 02104 - Remove virtual server (N)o/(y)es/(c)ontinue: y

Perform task: TASK: bigip_virtual_server : Issue 02104 - Remove virtual server (N)o/(y)es/(c)ontinue: ****************************************************************************

TASK [bigip_virtual_server : Issue 02104 - Remove virtual server] ****************************************************************************************************************
changed: [bigip1]

PLAY [Test the bigip_virtual_server module - Provider] ***************************************************************************************************************************
Perform task: TASK: Gathering Facts (N)o/(y)es/(c)ontinue: y

Perform task: TASK: Gathering Facts (N)o/(y)es/(c)ontinue: ***********************************************************************************************************************

TASK [Gathering Facts] ***********************************************************************************************************************************************************
ok: [bigip1]

PLAY RECAP ***********************************************************************************************************************************************************************
bigip1                     : ok=12   changed=4    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0

(venv-3.9) ➜  repro
(venv-3.9) ➜  repro ansible-playbook -i inventory/hosts ../f5-ansible/test/integration/bigip_virtual_server.yaml -t "issue-02104" --step
[WARNING]: Invalid characters were found in group names but not replaced, use -vvvv to see details

PLAY [Metadata of bigip_virtual_server] ******************************************************************************************************************************************

PLAY [Test the bigip_virtual_server module - Environment] ************************************************************************************************************************
Perform task: TASK: Gathering Facts (N)o/(y)es/(c)ontinue: y

Perform task: TASK: Gathering Facts (N)o/(y)es/(c)ontinue: ***********************************************************************************************************************

TASK [Gathering Facts] ***********************************************************************************************************************************************************
ok: [bigip1]
Perform task: TASK: bigip_virtual_server : Issue 02104 - Add virtual server (N)o/(y)es/(c)ontinue: y

Perform task: TASK: bigip_virtual_server : Issue 02104 - Add virtual server (N)o/(y)es/(c)ontinue: *******************************************************************************

TASK [bigip_virtual_server : Issue 02104 - Add virtual server] *******************************************************************************************************************
changed: [bigip1]
Perform task: TASK: bigip_virtual_server : Issue 02104 - Assert Add virtual server (N)o/(y)es/(c)ontinue: y

Perform task: TASK: bigip_virtual_server : Issue 02104 - Assert Add virtual server (N)o/(y)es/(c)ontinue: ************************************************************************

TASK [bigip_virtual_server : Issue 02104 - Assert Add virtual server] ************************************************************************************************************
ok: [bigip1] => {
    "changed": false,
    "msg": "All assertions passed"
}
Perform task: TASK: bigip_virtual_server : Issue 02104 - Add virtual server - Idempotent Check (N)o/(y)es/(c)ontinue: y

Perform task: TASK: bigip_virtual_server : Issue 02104 - Add virtual server - Idempotent Check (N)o/(y)es/(c)ontinue: ************************************************************

TASK [bigip_virtual_server : Issue 02104 - Add virtual server - Idempotent Check] ************************************************************************************************
ok: [bigip1]
Perform task: TASK: bigip_virtual_server : Issue 02104 - Assert Add virtual server - Idempotent Check (N)o/(y)es/(c)ontinue: y

Perform task: TASK: bigip_virtual_server : Issue 02104 - Assert Add virtual server - Idempotent Check (N)o/(y)es/(c)ontinue: *****************************************************

TASK [bigip_virtual_server : Issue 02104 - Assert Add virtual server - Idempotent Check] *****************************************************************************************
ok: [bigip1] => {
    "changed": false,
    "msg": "All assertions passed"
}
Perform task: TASK: bigip_virtual_server : Issue 02104 - Change auto last hop to enabled (N)o/(y)es/(c)ontinue: y

Perform task: TASK: bigip_virtual_server : Issue 02104 - Change auto last hop to enabled (N)o/(y)es/(c)ontinue: ******************************************************************

TASK [bigip_virtual_server : Issue 02104 - Change auto last hop to enabled] ******************************************************************************************************
changed: [bigip1]
Perform task: TASK: bigip_virtual_server : Issue 02104 - Assert Change auto last hop to enabled (N)o/(y)es/(c)ontinue: y

Perform task: TASK: bigip_virtual_server : Issue 02104 - Assert Change auto last hop to enabled (N)o/(y)es/(c)ontinue: ***********************************************************

TASK [bigip_virtual_server : Issue 02104 - Assert Change auto last hop to enabled] ***********************************************************************************************
ok: [bigip1] => {
    "changed": false,
    "msg": "All assertions passed"
}
Perform task: TASK: bigip_virtual_server : Issue 02104 - Change auto last hop to enabled - Idempotent Check (N)o/(y)es/(c)ontinue: y

Perform task: TASK: bigip_virtual_server : Issue 02104 - Change auto last hop to enabled - Idempotent Check (N)o/(y)es/(c)ontinue: ***********************************************

TASK [bigip_virtual_server : Issue 02104 - Change auto last hop to enabled - Idempotent Check] ***********************************************************************************
ok: [bigip1]
Perform task: TASK: bigip_virtual_server : Issue 02104 - Assert Change auto last hop to enabled - Idempotent Check (N)o/(y)es/(c)ontinue: y

Perform task: TASK: bigip_virtual_server : Issue 02104 - Assert Change auto last hop to enabled - Idempotent Check (N)o/(y)es/(c)ontinue: ****************************************

TASK [bigip_virtual_server : Issue 02104 - Assert Change auto last hop to enabled - Idempotent Check] ****************************************************************************
ok: [bigip1] => {
    "changed": false,
    "msg": "All assertions passed"
}
Perform task: TASK: bigip_virtual_server : Issue 02104 - Change auto last hop to disabled (N)o/(y)es/(c)ontinue: y

Perform task: TASK: bigip_virtual_server : Issue 02104 - Change auto last hop to disabled (N)o/(y)es/(c)ontinue: *****************************************************************

TASK [bigip_virtual_server : Issue 02104 - Change auto last hop to disabled] *****************************************************************************************************
changed: [bigip1]
Perform task: TASK: bigip_virtual_server : Issue 02104 - Assert Change auto last hop to disabled (N)o/(y)es/(c)ontinue: y

Perform task: TASK: bigip_virtual_server : Issue 02104 - Assert Change auto last hop to disabled (N)o/(y)es/(c)ontinue: **********************************************************

TASK [bigip_virtual_server : Issue 02104 - Assert Change auto last hop to disabled] **********************************************************************************************
ok: [bigip1] => {
    "changed": false,
    "msg": "All assertions passed"
}
Perform task: TASK: bigip_virtual_server : Issue 02104 - Change auto last hop to disabled - Idempotent Check (N)o/(y)es/(c)ontinue: y

Perform task: TASK: bigip_virtual_server : Issue 02104 - Change auto last hop to disabled - Idempotent Check (N)o/(y)es/(c)ontinue: **********************************************

TASK [bigip_virtual_server : Issue 02104 - Change auto last hop to disabled - Idempotent Check] **********************************************************************************
ok: [bigip1]
Perform task: TASK: bigip_virtual_server : Issue 02104 - Assert Change auto last hop to disabled - Idempotent Check (N)o/(y)es/(c)ontinue: y

Perform task: TASK: bigip_virtual_server : Issue 02104 - Assert Change auto last hop to disabled - Idempotent Check (N)o/(y)es/(c)ontinue: ***************************************

TASK [bigip_virtual_server : Issue 02104 - Assert Change auto last hop to disabled - Idempotent Check] ***************************************************************************
ok: [bigip1] => {
    "changed": false,
    "msg": "All assertions passed"
}
Perform task: TASK: bigip_virtual_server : Issue 02104 - Remove virtual server (N)o/(y)es/(c)ontinue: y

Perform task: TASK: bigip_virtual_server : Issue 02104 - Remove virtual server (N)o/(y)es/(c)ontinue: ****************************************************************************

TASK [bigip_virtual_server : Issue 02104 - Remove virtual server] ****************************************************************************************************************
changed: [bigip1]
Perform task: TASK: bigip_virtual_server : Issue 02104 - Assert Remove virtual server (N)o/(y)es/(c)ontinue: y

Perform task: TASK: bigip_virtual_server : Issue 02104 - Assert Remove virtual server (N)o/(y)es/(c)ontinue: *********************************************************************

TASK [bigip_virtual_server : Issue 02104 - Assert Remove virtual server] *********************************************************************************************************
ok: [bigip1] => {
    "changed": false,
    "msg": "All assertions passed"
}
Perform task: TASK: bigip_virtual_server : Issue 02104 - Remove virtual server - Idempotent Check (N)o/(y)es/(c)ontinue: y

Perform task: TASK: bigip_virtual_server : Issue 02104 - Remove virtual server - Idempotent Check (N)o/(y)es/(c)ontinue: *********************************************************

TASK [bigip_virtual_server : Issue 02104 - Remove virtual server - Idempotent Check] *********************************************************************************************
ok: [bigip1]
Perform task: TASK: bigip_virtual_server : Issue 02104 - Assert Remove virtual server - Idempotent Check (N)o/(y)es/(c)ontinue: y

Perform task: TASK: bigip_virtual_server : Issue 02104 - Assert Remove virtual server - Idempotent Check (N)o/(y)es/(c)ontinue: **************************************************

TASK [bigip_virtual_server : Issue 02104 - Assert Remove virtual server - Idempotent Check] **************************************************************************************
ok: [bigip1] => {
    "changed": false,
    "msg": "All assertions passed"
}

PLAY [Test the bigip_virtual_server module - Provider] ***************************************************************************************************************************
Perform task: TASK: Gathering Facts (N)o/(y)es/(c)ontinue: y

Perform task: TASK: Gathering Facts (N)o/(y)es/(c)ontinue: ***********************************************************************************************************************

TASK [Gathering Facts] ***********************************************************************************************************************************************************
ok: [bigip1]

PLAY RECAP ***********************************************************************************************************************************************************************
bigip1                     : ok=18   changed=4    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0

### CODE REVIEW

**Comment by RavinderReddyF5** [comment]
looks good to me

File: ansible_collections/f5networks/f5_modules/plugins/modules/bigip_virtual_server.py [hunk 1]
```diff
@@ -3676,6 +3702,10 @@ def __init__(self):
                 ]
             ),
             mirror=dict(type='bool'),
+            auto_last_hop=dict(
+                default='default',
```
Comment (wojtek0806) [inline ansible_collections/f5networks/f5_modules/plugins/modules/bigip_virtual_server.py:3706]: you might wanna remove default, if a user modifies a  VS which has this value set to something else, and the user does not specfiy it in their play, the module will override it on the device

**Review by wojtek0806 — CHANGES_REQUESTED** [review]
make the requested changes and retest after that it should be ok :)

**Comment by urohit011** [comment]
Hi @wojtek0806, I have made the changes you suggested.

**Review by wojtek0806 — APPROVED** [review]

**Pull request merged**